Wrangler2 release (#3909)
cloudflare/cloudflare-docs: content/workers/wrangler/migration/migrating-from-wrangler-1.md

---
pcx-content-type: how-to
title: Migrating from Wrangler 1
weight: 1
---

> ## Migrating from Wrangler 1

Reviewer: ```suggestion
## Migrate from Wrangler 1
```

> This document describes the steps to migrate a project from wrangler 1 to 2.

Reviewer: ```suggestion
This document describes the steps to migrate a project from Wrangler 1 to Wrangler 2.
```

> Wrangler version 2.0.0 ships with new features and improvements that may require some changes to your configuration.

Reviewer: ```suggestion
Wrangler 2 ships with new features and improvements that may require some changes to your configuration.
```

> The good news is that the CLI itself should guide you through the upgrade process.

Reviewer: Should guide you? As in sometimes it won't? Or is this just wording?

> The good news is that the CLI itself should guide you through the upgrade process.

Reviewer: ```suggestion
The CLI itself will guide you through the upgrade process.
```

{{<Aside type="node">}}

> To learn more about the improvements to Wrangler, refer to the [What's new page](/workers/wrangler/compare-v1-v2/).

Reviewer: Did we rename this page to Wranger updates? This needs updating

  - 😶 Ignored: "type":
    Most common features now work out of the box with wrangler, including modules, jsx,
  typescript, etc. If you need anything more, use a custom build.
  - Deprecation: "zone_id":
    This is unnecessary since we can deduce this from routes directly.
  - Deprecation: "build.upload.format":
    The format is inferred automatically from the code.

✘ [ERROR] Processing wrangler.toml configuration:
  - Expected "route" to be either a string, or an object with shape { pattern, zone_id | zone_name }, but got "".
```

### Common deprecations

Here are common fields that are no longer required.

- `type` is no longer required. Wrangler will infer the correct project type automatically.
- `zone_id` is no longer required. It can be deduced from the routes directly.
- `build.upload.format` is no longer used. The format is now inferred automatically from the code.
- `build.upload.main` and `build.upload.dir` are no longer required. Use the top level `main` field, which now serves as the entry-point for the Worker.
- `site.entry-point` is no longer required. The entry point should be specified through the `main` field.

> - `webpack_config` and `webpack` properties are no longer supported. See [Ejecting Webpack](/workers/wrangler/migration/eject-webpack/).

Reviewer: ```suggestion
- `webpack_config` and `webpack` properties are no longer supported. Refer to [Ejecting Webpack](/workers/wrangler/migration/eject-webpack/).
```

Here are the Wrangler 1 commands that are no longer supported:

- `wrangler preview` - Use the `wrangler dev` command, for running your worker in your local environment.
- `wrangler generate` - If you want to use a starter template, clone its GitHub repository and manually initialize it.
- `wrangler route` - Routes are defined in the `wrangler.toml` configuration file.
- `wrangler report` - If you find a bug please report it at [Wrangler issues](https://github.com/cloudflare/wrangler2/issues/new/choose).
- `wrangler build` - If you wish to access the output from bundling your Worker use `wrangler publish --outdir=path/to/output`.

> Refer to the [deprecations guide](/workers/wrangler/migration/deprecations/) for more details on what's no longer supported.

Reviewer: ```suggestion
Refer to the [deprecations guide](/workers/wrangler/migration/deprecations/) for more details on what is no longer supported.
```
